lib/graphql/query/literal_input.rb

```diff
@@ -86,7 +86,9 @@ def self.from_arguments(ast_arguments, argument_defns, variables)
           end
         end
 
-        GraphQL::Query::Arguments.new(values_hash, argument_definitions: argument_defns)
+        GraphQL::Query::Arguments
+          .construct_arguments_class(argument_definitions: argument_defns)
```

Comment:
Is this code called at runtime? I'm worried that we'll be creating classes on-the-fly instead of reusing ones that are already made. 

Maybe we should refactor `.from_arguments` to take different input. Instead of `argument_defns`, we should take `arguments_owner` (`GraphQL::Field` or `GraphQL::InputObjectType`) then we can get the arguments class from there. (Disclaimer: I don't remember _exactly_ how this method is used so it might not work!!)

Reply:
Yeah, I'm pretty sure this one's in runtime. There didn't seem to be an easy way to improve that, so I might need some help to look into that :grin: